### PR TITLE
Added Aff interface

### PR DESCRIPTION
**Pull request by cryogenian**

### CODE REVIEW

**Comment by jdegoes** [comment]
This code is not asynchronous, is it? Why not use `liftEff` instead?

**Comment by cryogenian** [comment]
`matches` is asynchronous since it event handler. 
I've added async setters because they won't break anything and there is dependency on `purescript-aff` if `matchesAff` added

**Comment by jdegoes** [comment]
Ah, OK, I didn't see that. :+1: Looks good to me, @garyb any comments before merging?

File: src/Routing/Hash/Aff.purs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import Control.Monad.Aff
+import qualified Routing.Hash as R
+
+modifyHash :: forall e. (String -> String) -> Aff (dom :: DOM|e) Unit 
+modifyHash func = makeAff \_ k -> do 
+  R.modifyHash func
+  k unit
+
+setHash :: forall e. String -> Aff (dom :: DOM|e) Unit 
+setHash hash = makeAff \_ k -> do
```
Comment (jdegoes) [inline src/Routing/Hash/Aff.purs]: It's easier to use `liftEff` here.

File: src/Routing/Hash/Aff.purs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+module Routing.Hash.Aff where
+
+import DOM 
+import Control.Monad.Aff
+import qualified Routing.Hash as R
+
+modifyHash :: forall e. (String -> String) -> Aff (dom :: DOM|e) Unit 
+modifyHash func = makeAff \_ k -> do 
+  R.modifyHash func
```
Comment (jdegoes) [inline src/Routing/Hash/Aff.purs]: It's easier (and faster) to use `liftEff` here.

**Comment by cryogenian** [comment]
switched to `liftEff` in `Routing.Hash.Aff`

**Comment by garyb** [comment]
:shipit:

**Pull request merged**